### ISSUE TITLE
[Spike] Investigate rhtap nightly devfile compatibility check failures

### ISSUE THREAD

**Issue by thepetk** [devfile/api#1414]
## Which area/kind this issue is related to?

/area ci
/area registry
/area test-automation

## Issue Description

This issue aims to investigate the reason of the failing `rhtap-nightly-devfile-compatibility-check`. More detailed, during the last week (Tuesday 2nd Jan 2024 -> Monday 8th Jan 2024) half of the returned a failure (50%).

An example trace can be found here: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-devfile-registry-main-v4.12-rhtap-nightly-devfile-compatibility/1744328191855562752

## Acceptance Criteria

- [ ] Share investigation results and create required issues

**Comment by thepetk** [comment]
I've noticed that the mast majority last 13 jobs have failed. I've digged a little bit and I found that pretty much all of them failed with the following error trace:
```
NoClusters: No clusters in pool are ready to be claimed
```

The [application service configuration](https://github.com/flacatus/release/blob/master/ci-operator/config/redhat-appstudio/application-service/redhat-appstudio-application-service-main.yaml) in the openshift/release repo has been updated ([#44969](https://github.com/openshift/release/pull/44969/files)) to use cluster pools. The `cluster_claim.timeout` is set to `1h`.

Looking into other configs (example [here from appsre](https://github.com/openshift/release/blob/master/ci-operator/config/app-sre/deployment-validation-operator/app-sre-deployment-validation-operator-master.yaml#L68-L74)) I've noticed that some configs use bigger timeout for the cluster claim.

As a result I think it might be a good start to increase the time out (double it to `2h`) and monitor the job to see if this has allocated more time to find a cluster for the job.
